sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -288,7 +288,7 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       val provider = maybeV2Provider.get
       val sessionOptions = DataSourceV2Utils.extractSessionConfigs(
         provider, df.sparkSession.sessionState.conf)
-      val options = sessionOptions ++ extraOptions
```

Comment:
maybe `sessionOptions.filterNot(extraOptions.contains) ++ extraOptions.originalMap`

Reply:
Sure. I'll use `sessionOptions.filterKeys(!extraOptions.contains(_)) ++ extraOptions.originalMap`.

Reply:
actually, `originalMap` should be `toMap`

Reply:
Oh, got it.